### DISCUSSION TITLE
If using strategy 'jwt' do I need to implement full adapter for "email" provider?

### DISCUSSION THREAD

**Discussion by tombburnell** [nextauthjs/next-auth#4430] [Help]
I've been using github and facebook providers and just JWT for sessions. 

I want to add the email provider now.  From reading the docs it looks like I have to provide an Adapter to satisfy the verification token endpoints. I created those but I'm getting errors that I need to implement getUser and updateUser.

Do I have to implement the full Adapter spec and model to go with it?

Thanks

**Comment by ivan-kleshnin** [comment]
For JWT-based sessions you can certainly omit (mock) the following:

```
    async createSession(_session: {sessionToken: string, userId: string, expires: Date}) {
      return null as any
    },

    async updateSession(_session: Partial<AdapterSession> & Pick<AdapterSession, "sessionToken">) {
      return null
    },

    async deleteSession(_sessionToken: string) {
      return null
    },

    async getSessionAndUser(_sessionToken: string) {
      return null
    }
```

`updateUser` is necessary because it will set `emailVerified` to `true`. 
In short, I'd personally implement other 6 "required" methods.
